tests/Unit/Generator/EntityRequestSchemaBuilderTest.php

```diff
@@ -0,0 +1,170 @@
+    {
+        $factory = $this->createMock(ClassMetadataFactory::class);
+        $factory->expects(self::once())
+            ->method('isTransient')
+            ->with(PublicProperties::class)
+            ->willReturn(false);
+
+        $manager = $this->createMock(ObjectManager::class);
+        $manager->expects(self::once())
+            ->method('getMetadataFactory')
+            ->willReturn($factory);
+
+        $managerRegistry = new ManagerRegistryStub([
+            PublicProperties::class => $manager,
+        ]);
+
+        $builder = new EntityRequestSchemaBuilder($managerRegistry);
+
+        $expected = new Schema([
+            'properties' => [
+                'id' => new Schema([
+                    'type' => 'string',
+                ]),
+            ],
+        ]);
+
+        $actual = $builder->buildSchema(PublicProperties::class);
+
+        self::assertEquals($expected, $actual);
+    }
+
+    /**
+     * Tests that supports listens to its internal $skipEntities array.
+     *
+     * @return void
+     */
+    public function testSupportsSkips(): void
+    {
+        $managerRegistry = new ManagerRegistryStub();
+        $builder = new EntityRequestSchemaBuilder(
+            $managerRegistry,
+            null,
+            [PublicProperties::class]
+        );
+
+        $actual = $builder->supports(PublicProperties::class);
+
+        self::assertFalse($actual);
+    }
+
+    /**
+     * Tests that supports listens to its internal $skipEntities array.
+     *
+     * @return void
+     */
+    public function testSupportsFalseWhenNoEMReturned(): void
+    {
+        $managerRegistry = new ManagerRegistryStub();
+        $builder = new EntityRequestSchemaBuilder($managerRegistry);
+
+        $actual = $builder->supports(PublicProperties::class);
+
+        self::assertFalse($actual);
+    }
+
+    /**
+     * Tests that supports listens to its internal $skipEntities array.
+     *
+     * @return void
+     */
+    public function testSupportsFalseWhenTransient(): void
+    {
+        $factory = $this->createMock(ClassMetadataFactory::class);
+        $factory->expects(self::once())
+            ->method('isTransient')
+            ->with(PublicProperties::class)
+            ->willReturn(true);
+
+        $manager = $this->createMock(ObjectManager::class);
+        $manager->expects(self::once())
+            ->method('getMetadataFactory')
+            ->willReturn($factory);
+
+        $managerRegistry = new ManagerRegistryStub([
+            PublicProperties::class => $manager,
+        ]);
+        $builder = new EntityRequestSchemaBuilder($managerRegistry);
+
+        $actual = $builder->supports(PublicProperties::class);
+
+        self::assertFalse($actual);
+    }
+
+    /**
+     * Tests that supports listens to its internal $skipEntities array.
+     *
+     * @return void
+     */
+    public function testSupports(): void
+    {
+        $factory = $this->createMock(ClassMetadataFactory::class);
```

Comment:
I really don't like the use of mocks unless it's really necessary, they mask errors which we find when we stub. This is why they're generally removed.

Reply:
Happy to discuss this one but I believe the standard is no mocks unless there is absolutely no other option.